test/karma.conf.js

```diff
@@ -150,3 +157,21 @@ module.exports = (config) => {
     config.browsers = platformMap[platform];
   }
 };
+
+/**
+ * Splits the specified browsers into a maximum amount of chunks. The chunk of browsers
+ * are being created deterministically and therefore we get reproducible tests when executing
+ * the same CircleCI instance multiple times.
+ */
+function splitBrowsersIntoInstances(browsers, maxInstances) {
+  let chunks = [];
+  let assignedBrowsers = 0;
+
+  for (let i = 0; i < maxInstances; i++) {
+    const chunkSize = Math.floor((browsers.length - assignedBrowsers) / (maxInstances - i));
+    chunks[i] = browsers.slice(assignedBrowsers, assignedBrowsers + chunkSize);
+    assignedBrowsers += chunkSize;
+  }
```

Comment:
@jelbourn You probably have some better ideas to split an array into a maximum size of chunks in a deterministic way.

Note that I intentionally didn't consider splitting the actual tests across browsers because we want to run _all specs_ on all those browsers.